### PR TITLE
Ts EBI header

### PR DESCRIPTION
**Pull request by gustavo-salazar**
Typescript and visual framework migration of EBIHeader.

There is a bit of redundancy on styles with the `EBI_global.css` that is still loaded in the website until we completely drop EBI framework 1.4.

The new story in storybook shows the header loading using only the `vf-` classes.

Also removing the EMBLDropdown component that hasn't been used for a while.

### CODE REVIEW

**Review by matthiasblum — CHANGES_REQUESTED** [review]
Thank you.

The items of the header are all on the right-hand side of the page. I would expect them to be inside a container with a `max-width`:

`dev` branch:

image

`ts-ebi-header` branch:

image

---

Also, I am not sure _how_, but the changes seem to be responsible for style changes _outside_ the header.

`dev` branch:

client-dev

The **InterPro** header is rendered using Ubuntu Bold (on my system), due to `font-weight: 600`.

`ts-ebi-header` branch:

client-ts-ebiheader

Same header is now rendered with `font-weight: 400`. The main body/container of the page is a bit wider on `dev`. I believe both changes are due to Foundation's rules taking priority on  `ts-ebi-header`. But why or how, I have no idea.

**Comment by gustavo-salazar** [comment]
OK, so the `global_ebi.css` includes redefinitions for high-level elements (e.g. `<body>`). On the migration to VF, we are not including that for the header anymore. I thought that wouldn't be a problem because it is included elsewhere. However, there are other files defining the same high-level objects(e.g. foundation).
And given that the CSS specificity of the files was the same (the selector in all of them is `body {`), it comes down to which one is the last one to be included. In the case of files, it comes down to what gets loaded first, making it a bit random, so sometimes the headers are defined by one file and other times by another. An unfortunate consequence of this transitional state between frameworks.

My solution to start detaching any dependency on `global_ebi.css` is to start defining these values in `interpro-vf.css` but also add a class to the `body` so the selector there has more specificity. 
This might have side-effects on particular styling of some headers(e.g. subtitle in main banner), and we will need to keep an eye for such cases

File: src/components/EBIHeader/style.css [hunk 1]
```diff
@@ -1,25 +1,68 @@
...
-    float: right !important;
+        & > a {
+          &,
+          &:visited,
+          &:focus {
+            color: white;
+          }
+          &:not(.float-right):hover {
+            color: #373a36;
+            background-color: #d0d0ce !important;
```
Comment (matthiasblum) [inline src/components/EBIHeader/style.css:22]: I don't know why I originally didn't use the custom properties.

```suggestion
            color: var(--vf-color--grey--darkest);
            background-color: var(--vf-color--grey--lightest);
```

File: src/components/EBIHeader/style.css [hunk 2]
```diff
@@ -1,25 +1,68 @@
...
 /* Overwrite EBI guidelines styles… */
-.global-nav > * {
-  float: initial !important;
+.masthead-black-bar {
+  --vf-stack-margin--custom: 0;
+  background-color: var(--vf-color--grey--darkest);
+  /* border-top: 0 solid #eee; */
+
+  & > nav {
+    max-width: var(--vf-body-width, 80em);
```
Comment (matthiasblum) [inline src/components/EBIHeader/style.css:8]: Center the `nav`:

```suggestion
    max-width: var(--vf-body-width, 80em);
    margin: 0 auto;
```

**Review by matthiasblum — CHANGES_REQUESTED** [review]
Thanks for finding the issue.

I made two small suggestions to the CSS file, and there are a few lines that can be removed in `index.template.html`:

 ```css
      .tmp-ebi-header {
        background-color: #373a36;
        height: 37.289px;
      }
```

**Comment by gustavo-salazar** [comment]
> I made two small suggestions to the CSS file, 

Cool, merged them.

> and there are a few lines that can be removed in `index.template.html`:
> 
> ```css
>      .tmp-ebi-header {
>        background-color: #373a36;
>        height: 37.289px;
>      }
> ```

This actually has a purpose: the idea is that it uses this minimal CSS to create the block for the banner and show something while all the react components get loaded, so there is a minimal skeleton of the page instead of a blank page.

**Comment by matthiasblum** [comment]
> This actually has a purpose: the idea is that it uses this minimal CSS to create the block for the banner and show something while all the react components get loaded, so there is a minimal skeleton of the page instead of a blank page.

I see. Then maybe the `height` rule could be removed? `37.289px` is more than the max height of nav bar items.

**Review by matthiasblum — APPROVED** [review]

**Comment by gustavo-salazar** [comment]
> I see. Then maybe the `height` rule could be removed? `37.289px` is more than the max height of nav bar items.

Nope. It is exactly the height that will be displayed. So when just the HTML gets loaded we got this:
Screenshot 2024-05-01 at 09 58 27

And when React takes over we get this:

Screenshot 2024-05-01 at 09 58 00

**Pull request merged**